run.py

```diff
@@ -86,11 +86,22 @@ def checkerboard_crossover(candidate1: List[List[Intersection]], candidate2: Lis
                 new_candidate[y].append(candidate2[y][x])
     return new_candidate
 
+def uniform_crossover(candidate1: List[List[Intersection]], candidate2: List[List[Intersection]]):
+    # builds a new candidate by randomly alternating between either parent cell by cell
+    new_candidate = []
+    for y in range(len(candidate1)):
+        new_candidate.append([])
+        for x in range(len(candidate1[0])):
+            new_candidate[y].append(candidate1[y][x] if random.random() < 0.5 else candidate2[y][x])
+    return new_candidate
+
 def crossover(candidate1: List[List[Intersection]], candidate2: List[List[Intersection]]):
-    if random.random() < 0.5:
+    if random.random() < 0.33:
         return checkerboard_crossover(candidate1, candidate2)
-    else:
+    elif random.random() < 0.66:
```

Comment:
if you want all three branches to have .33 chance, i think you need to assign the random.random() to a variable before. this actually gives more than > .33 chance to the middle one and < .33 chance to the last one.